File: tests/fixtures/smoke/ruby/sample.rb
# Smoke-tier fixture (BACK-431 Issue G) — Ruby.
def validate(order)
  raise ArgumentError, "empty order" if order.nil?
  order
end

def process_order(order)
  result = validate(order)
  count = 0
  begin
    while count < 3
      count += 1
    end
    result.upcase
  rescue StandardError
    nil
  end
end

def run(order)
  process_order(order)
end

File: tests/fixtures/smoke/dart/sample.dart
// Smoke-tier fixture (BACK-431 Issue G) — Dart.
String validate(String? order) {
  if (order == null) {
    throw ArgumentError("empty order");
  }
  return order;
}

String? processOrder(String? order) {
  try {
    final result = validate(order);
    var count = 0;
    while (count < 3) {
      count += 1;
    }
    return result.toUpperCase();
  } catch (e) {
    return null;
  }
}

String? run(String? order) {
  return processOrder(order);
}
